elytron/WFCORE-4496-principal-transformer-in-aggregate-realm.adoc

```diff
@@ -0,0 +1,60 @@
+= WFCORE-4496 Adding principal-transformer in aggregate-realm
+:author:            Ashley Abdel-Sayed
+:email:             aabdelsa@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+Elytron currently has an aggregate security realm which is a combination of two realms; an authentication and an
+authorization realm. There is currently no way to transform the principal in between the two realms.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-4496[WFCORE-4496]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1220[EAP7-1220]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+* mailto:vmarek@redhat.com[Vratislav Marek]
+
+//=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+//[ ] Engineering
+//[ ] QE
+
+=== Affected Projects or Components
+* WildFly, Security
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* It should be possible to configure an ```aggregate-realm``` with the optional attribute ```principal-transformer``` which will
+define transformations to the principal after the authentication-realm identity is generated but before the authorization-realm
+identity is generated.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+== Test Plan
+Subsystem parsing and transformer tests will be added. Tests will also be added to the Elytron testsuite and the Elytron
+subsystem tests to ensure that the principal is correctly transformed after authentication and before authorization.
+
+//== Community Documentation
+////
+Generally a feature should have documentation as part of the PR to wildfly master, or as a follow up PR if the feature is in wildfly-core. In some cases though the documentation belongs more in a component, or does not need any documentation. Indicate which of these will happen.
```

Comment:
It would be good to mention where the documentation for this new attribute will be added.

Reply:
Added